### PR TITLE
Mobile: Remove logs

### PR DESCRIPTION
**Pull request by Tonksthebear**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Removes logging the team doesn't use any more

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7930

## Testing done
Removes testing code related to the logs missing

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by Tonksthebear** [comment]
I left all secure messaging logs as is seems that's a point of interest still. Also I haven't removed the tests yet until we decide which logs are worth removing

File: modules/mobile/app/services/mobile/v0/legacy_disability_rating/proxy.rb [hunk 1]
```diff
@@ -29,7 +29,6 @@ def get_disability_ratings # rubocop:disable Metrics/MethodLength
           raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error'
         rescue => e
           if e.respond_to?('response')
-            Rails.logger.info('LEGACY DR ERRORS WITH RESPONSE', error: e)
```
Comment (Tonksthebear) [inline modules/mobile/app/services/mobile/v0/legacy_disability_rating/proxy.rb:32]: both of these logs are followed by an error so worst case we can look up the error

File: modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb [hunk 2]
```diff
@@ -409,14 +406,6 @@ def parse_phone(phone)
             }
           end
 
-          unless phone.nil?
-            # this logging is intended to be temporary. Remove if it's not occurring
```
[the remark this answers is not in the document]
Comment (Tonksthebear) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:413]: Wow we are very much still logging this. Will leave

**Review by aherzberg — DISMISSED** [review]
LGTM

**Review by aherzberg — APPROVED** [review]
LGTM

**Pull request merged**